src/globemap.ts

```diff
@@ -721,33 +719,104 @@ module powerbi.extensibility.visual {
+                return null;
+
+            tileCacheArray.forEach((obj) => {
+                for (let key in obj) {
+                    if (obj.hasOwnProperty(key)) {
+                        const mainInfo = obj[key];
+                        obj[key] = `${protocol}${lastDomain}.${mainInfo.domain}.${firstDomains}/tiles/${mainInfo.file}.${extension}?${additionsParams}`;
+                    }
+                }
+            });
+
+            return tileCacheArray;
+        }
+
+        private setBingMetadata(): JQueryPromise<any> {
+            return this.getBingMapsServerMetadata()
+                .then((metadata: BingResourceMetadata) => {
+                    let tileCacheValue = [];
+                    let urlTemplate = metadata.imageUrl.replace("{culture}", this.currentLanguage);
+                    for (let level: number = GlobeMap.initialResolutionLevel; level <= GlobeMap.maxResolutionLevel; ++level) {
+                        let levelTiles = this.generateQuadsByLevel(level, urlTemplate, metadata.imageUrlSubdomains);
+                        this.mapTextures.push(this.createTexture(level, levelTiles));
+                        tileCacheValue.push(levelTiles);
+                    }
+
+                    const minimizedTileCacheData: string = JSON.stringify(this.minimizeBingMetadata(tileCacheValue));
+                    this.localStorageService.set(GlobeMap.TILE_STORAGE_KEY, minimizedTileCacheData);
+                    this.localStorageService.set(GlobeMap.TILE_LANGUAGE_CULTURE, this.currentLanguage);
+                    return tileCacheValue;
+                });
+        }
+
+        private setDataToLocalStorage(tileCulture: string, tileCacheValueObj: any): JQueryPromise<{}> {
+
+            if (!tileCacheValueObj || tileCulture !== this.currentLanguage) {
+                this.setBingMetadata();
             } else {
+                const titleCacheExtended = this.extendBingMetadata(tileCacheValueObj);
+                console.log("tileCacheValueObj: ", titleCacheExtended);
                 for (let level: number = GlobeMap.initialResolutionLevel; level <= GlobeMap.maxResolutionLevel; ++level) {
-                    this.mapTextures.push(this.createTexture(level, tileCache[level - GlobeMap.initialResolutionLevel]));
+                    this.mapTextures.push(this.createTexture(level, titleCacheExtended[level - GlobeMap.initialResolutionLevel]));
                 }
-                return jQuery.when(tileCache);
+                return jQuery.when(titleCacheExtended);
             }
         }
 
+        private initTextures(): JQueryPromise<{}> {
+            this.mapTextures = [];
+            const tileCulturePromise: IPromise<string> = this.localStorageService.get(GlobeMap.TILE_LANGUAGE_CULTURE);
+            let tileCachePromise: IPromise<string> = this.localStorageService.get(GlobeMap.TILE_STORAGE_KEY);
+            let deferred = $.Deferred();
+
+            tileCulturePromise.then((tileCultureValue: string) => {
```

Comment:
Why tileCacheValues depend on tileCultureValue?  Can we storage tileCacheValues names with CultureKey?